opengrok-tools module breaks the maven build under Windows

At commit c1e38f7b.

Issues are: with default python install, mvn -DskipTests clean install:
```
[main] [INFO] OpenGrok tools ..................................... FAILURE [  2.190 s]
[main] [INFO] Distribution ....................................... SKIPPED
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] BUILD FAILURE
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] Total time: 2.323 s
[main] [INFO] Finished at: 2018-10-29T14:15:36+13:00
[main] [INFO] Final Memory: 25M/378M
[main] [INFO] ------------------------------------------------------------------------
[main] [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (generate python env) on project opengrok-tools: Command execution failed.: Cannot run program "python3" (in directory "C:\java\opengrok\buildfromsource\OpenGrok\opengrok-tools\target"): CreateProcess error=2, The system cannot find the file specified -> [Help 1]
```
I see from the readme.md that the scripts require a python3 symlink to run, but should it be required to build as well? Even with a python3.cmd alias to python.exe:
```
[main] [INFO] Reactor Summary:
[main] [INFO]
[main] [INFO] OpenGrok tools ..................................... FAILURE [ 11.345 s]
[main] [INFO] Distribution ....................................... SKIPPED
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] BUILD FAILURE
[main] [INFO] ------------------------------------------------------------------------
[main] [INFO] Total time: 11.499 s
[main] [INFO] Finished at: 2018-10-29T14:16:22+13:00
[main] [INFO] Final Memory: 25M/372M
[main] [INFO] ------------------------------------------------------------------------
[main] [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (generate-package) on project opengrok-tools: Command execution failed.: Cannot run program "env\bin\python" (in directory "C:\java\opengrok\buildfromsource\OpenGrok\opengrok-tools\target"): CreateProcess error=2, The system cannot find the file specified -> [Help 1]
```
The issue here is that the virtual envs that Python creates are different on Windows, with an env\Scripts folder instead of env/bin.

I can probably fix this with OS-specific Maven profiles for the different paths, is that OK?

--- a/opengrok-tools/pom.xml
+++ b/opengrok-tools/pom.xml
@@ -38,6 +38,34 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
 
     <name>OpenGrok tools</name>
 
+    <profiles>
+        <profile>
+            <id>not-windows</id>
+            <activation>
+                <os>
+                    <family>!windows</family>
+                </os>
+            </activation>
+            <properties>
+                <python3Command>python3</python3Command>
+                <pythonEnvBin>env/bin</pythonEnvBin>
+            </properties>
+        </profile>
+        <profile>
+            <id>windows</id>
+            <activation>
+                <os>
+                    <family>windows</family>
+                </os>
+            </activation>
+            <properties>
+                <python3Command>python</python3Command>
+                <pythonEnvBin>env/Scripts</pythonEnvBin>
+                <!-- Tests do not run on Windows because the python code is Unix-specific -->
+                <skipTests>true</skipTests>
+            </properties>
+        </profile>
+    </profiles>
     <build>
         <sourceDirectory>src/main/python</sourceDirectory>
         <testSourceDirectory>src/test/python</testSourceDirectory>
@@ -163,7 +191,7 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                     <execution>
                         <id>generate python env</id>
                         <configuration>
-                            <executable>python3</executable>
+                            <executable>${python3Command}</executable>
                             <workingDirectory>${project.build.directory}</workingDirectory>
                             <arguments>
                                 <argument>-m</argument>
@@ -180,7 +208,7 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                         <!-- needed for 'pip install pylint' to work -->
                         <id>Upgrade pip</id>
                         <configuration>
-                            <executable>env/bin/pip3</executable>
+                            <executable>${pythonEnvBin}/pip3</executable>
                             <workingDirectory>${project.build.directory}</workingDirectory> 
                             <arguments>
                                 <argument>install</argument>
@@ -200,11 +228,13 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                             <goal>exec</goal>
                         </goals>
                         <configuration>
-                            <executable>env/bin/python</executable>
+                            <executable>${pythonEnvBin}/python</executable>
                             <workingDirectory>${project.build.directory}</workingDirectory>
                             <arguments>
                                 <argument>setup.py</argument>
                                 <argument>sdist</argument>
+								<argument>--formats</argument>
+								<argument>gztar</argument>
                             </arguments>
                         </configuration>
                         <phase>package</phase>
@@ -212,7 +242,7 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                     <execution>
                         <id>python-test-install</id>
                         <configuration>
-                            <executable>env/bin/python</executable>
+                            <executable>${pythonEnvBin}/python</executable>
                             <workingDirectory>${project.build.directory}</workingDirectory>
                             <arguments>
                                 <argument>setup.py</argument>

